Source Code/Ticket Management System/supportgiver_menu.c

```diff
@@ -77,143 +77,157 @@ bool searchByTags(void* ticket, void* tag)
-						struct Vector* filtered_vector = vector_filter(tickets, filterByStakeholders, stakeholders);
-						printTicketsArray(filtered_vector);
-						vector_destroy(stakeholders_vector);
-						vector_destroy(filtered_vector);
-						break;
-					}
-					case('5'): // Search by customer email
-					{
-						wchar_t* ctx;
-						wchar_t customer_email[32];
-						wprintf(L"%s\n", L"please input Customer Email");
-						wscanf(L"%s", customer_email);
-						wcstok(customer_email, "\n", &ctx);
-						struct Vector* filtered_vector = vector_filter(tickets, searchByCustomerEmail, customer_email);
-						printTicketsArray(filtered_vector);
-						vector_destroy(filtered_vector);
-						break;
-					}
-					case('6'): // Search by tags
-					{
-						wchar_t* ctx;
-						wchar_t tags[32];
-						wprintf(L"%s\n", L"please input tags");
-						wscanf(L"%s", tags);
-						wcstok(tags, "\n", &ctx);
-						struct Vector* filtered_vector = vector_filter(tickets, searchByTags, tags);
-						printTicketsArray(filtered_vector);
-						vector_destroy(filtered_vector);
-						break;
-						
-					}
-					case('7'): // Sort by title
-					{
-						struct Vector* sorted_vector = vector_sort(tickets, sortByTitle);
-						printTicketsArray(sorted_vector);
-						vector_destroy(sorted_vector);
-						break;
-					}
-					case('q'):
-					case('Q'):
-					{
-						go_back = true;
-						break;
-					}
-					default:
-					{
-						input_is_valid = false;
-						wprintf(L"Input is invalid. Try again.\n");
-					}
-				}
-			} while (!input_is_valid);
-		}
+		//Creating the menu and giving it a title
+		struct Menu* browse_tickets = menu_create();
+		menu_setTitle(browse_tickets, L"Browse Tickets\n==============");
+
+		//Creating the options and inserting them into the menu
+		struct MenuOption* statusOption = menuOption_create(L"Filter by status", filterByStatusAction);
+		struct MenuOption* tierOption = menuOption_create(L"Filter by tier", filterByTierAction);
+		struct MenuOption* typeOption = menuOption_create(L"Filter by type", filterByTypeAction);
+		struct MenuOption* stakeholdersOption = menuOption_create(L"Filter by stakeholders", filterByStakeHolderAction);
+		struct MenuOption* emailOption = menuOption_create(L"Search by customer email", SearchByCustomerEmailAction);
+		struct MenuOption* tagsOption = menuOption_create(L"Search by tags", SearchByTagsAction);
+		struct MenuOption* titleSortOption = menuOption_create(L"Sort by title", SortByTitleAction);
+
+		//Inserting the options into "browse_tickets" menu struct
+		menu_addOption(browse_tickets, statusOption);
+		menu_addOption(browse_tickets, tierOption);
+		menu_addOption(browse_tickets, typeOption);
+		menu_addOption(browse_tickets, stakeholdersOption);
+		menu_addOption(browse_tickets, emailOption);
+		menu_addOption(browse_tickets, tagsOption);
+		menu_addOption(browse_tickets, titleSortOption);
+
+		//Displaying the menu "browse_tickets"
+		menu_tick(browse_tickets);
```

Comment:
What happens when the user inputs an invalid option?
They get sent to the main menu.